src/engine/drpc_client.c

```diff
@@ -390,7 +390,12 @@ ds_pool_find_bylabel(d_const_string_t label, uuid_t pool_uuid,
 	if (ranks == NULL)
 		D_GOTO(out_resp, rc = -DER_NOMEM);
 	*svc_ranks = ranks;
-	uuid_parse(frsp->uuid, pool_uuid);
+	rc = uuid_parse(frsp->uuid, pool_uuid);
+	if (rc != 0) {
+		D_ERROR("Unable to parse pool UUID %s: "DF_RC"\n", frsp->uuid,
+			DP_RC(rc));
+		D_GOTO(out_resp, rc = -DER_INVAL);
```

Comment:
Generally, if a function like ds_pool_find_bylabel returns an error, its caller is not expected to look at pool_uuid and *svc_ranks. So we should free ranks before returning the -DER_INVAL. E.g., something like

    rc = uuid_parse(...);
    if (rc != 0) {
        ...
        D_FREE(*svc_ranks);
        D_GOTO(...);
    }

Additionally, is -DER_INVAL really appropriate? The caller hasn't specified any invalid argument in this case; it's the control plane supplying an invalid response...

Reply:
Or better only assign ranks to *svc_ranks after frsp->uuid has been successfully parsed.